### PR TITLE
🐞 fix(text): adjust prop size (deprecated) to fontSize

### PR DESCRIPTION
**Pull request by oharaju**
[JIRA Issue](https://github.com/gympass/yoga/issues/590)

## Description 📄

To change the font size it is necessary to add a prop called size, but when declaring it I noticed that the prop does not work very well in the <Text>.

The Text component has compatibility with the [Yoga System](https://gympass.github.io/yoga/system/getting-started), using the [typographic resources](https://gympass.github.io/yoga/system/typography), which is duplicated with regard to the properties to change the font size. The size property has been deprecated by Yoga System (typography), but the documentation has not been updated.

This PR changes the size property to `fontSize` of the `Text` component to fix the incompatibility issue.

**More details in issue [#590](https://github.com/gympass/yoga/issues/590).**

### Changes

- Documentation adjustments;
- Snapshot test update.

## Platforms 📲

- [x] Web
- [x] Mobile

## Type of change 🔍

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested? 🧪

- [ ] Unit Test
- [x] Snapshot Test

## Checklist: 🔍

- [x] My code follows the contribution guide of this project [Contributing Guide](https://github.com/Gympass/yoga/blob/master/CONTRIBUTING.md)
- [ ] Layout matches design prototype: [FIGMA](https://figma.com/file/YOUR_LINK)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Screenshots 📸

| Before | After |
| ------ | ----- |
| image |  image |

### CODE REVIEW

File: packages/yoga/src/Text/native/Text.jsx [hunk 1]
```diff
@@ -74,6 +74,7 @@ Text.displayName = 'Text';
 
 Text.propTypes = {
   inverted: bool,
+  fontSize: oneOfType([string, number]),
```
Comment (caiotracera) [inline packages/yoga/src/Text/native/Text.jsx:77]: Maybe we should add the accepted values for this prop, just like we did on `size` prop. My suggestion:

```suggestion
  fontSize: oneOfType([
      'xxsmall',
      'xsmall',
      'small',
      'medium',
      'large',
      'xlarge',
      'xxlarge',
      'xxxlarge',
      'huge',
    ]),
```

What do you think?

File: packages/yoga/src/Text/web/Text.jsx [hunk 2]
```diff
@@ -80,6 +80,7 @@ Text.displayName = 'Text';
 Text.propTypes = {
   inverted: bool,
   numberOfLines: number,
+  fontSize: oneOfType([string, number]),
```
Comment (caiotracera) [inline packages/yoga/src/Text/web/Text.jsx:83]: Same here regarding the declaration of the accepted values for this prop

**Review by caiotracera — COMMENTED** [review]
Hey, @oharaju! Firstly, congrats for your first pull request 🎉🥳
The code looks good to me, but I added a suggestion about adding the accepted values to the `fontSize` property. Let me know your opinion about that!

**Comment by matheushdsbr** [comment]
Yo @oharaju !! o/

I am very happy with your contribution! 😁 
These points you raised are very interesting. I also saw that you opened a issue talking about that problem, very cool! 
The entire Yoga team appreciates your interest in contributing to our project! 🚀🚀

File: packages/yoga/src/Text/native/Text.jsx (hunk 1 above)
Comment (oharaju) [reply packages/yoga/src/Text/native/Text.jsx]: Hi, @caiotracera! First I would like to thank you for the review, I am very happy for your attention. 🚀🎉

Interesting! On second thought, it really makes sense to add the values accepted by the prop.

File: packages/yoga/src/Heading/web/__snapshots__/Heading.test.jsx.snap [hunk 3]
```diff
@@ -117,10 +117,11 @@ exports[`<Heading /> should match snapshot no padding 1`] = `
 .c5 {
   margin: 0;
   padding: 0;
-  font-size: 14px;
+  font-size: 16px;
   font-weight: 500;
   font-family: Rubik;
   color: #231B22;
+  font-size: 14px;
```
Comment (ericcleao) [inline packages/yoga/src/Heading/web/__snapshots__/Heading.test.jsx.snap:124]: @oharaju, first of all great contribution! 🎉
Looking at this snapshot test it looks like that when the `size` prop at the test file was being used the `font-size` on the style was set to `14px` directly but using `font-size` the style looks like to be setting it to `16px` and after being updated to `14px`.
Could we check this behaviour? Another fact that I checked is that any mobile snapshot changed the style after the change from `size` your `fontSize` it seems to work just the same.

**Review by ericcleao — COMMENTED** [review]
@oharaju @caiotracera @matheushdsbr I'm trying to check if it was really a bug or expected behaviour, because it seems that for higher sizes the `size` prop was working fine, maybe the propTypes were leading the developer to use not accepted fontSizes (too small sizes).
What do you think?

**Comment by caiotracera** [comment]
Hey, @ericcleao!
Can you give us some examples on what you said?

**Comment by caiotracera** [comment]
Hey, @oharaju!
We (me, @matheushdsbr and @caioalexandrebr) had an internal meeting to discuss your PR and the points raised by @ericcleao. For now, we decided the following:

- Just add the `deprecated` to `size` prop on the `Text` component, as you did. Doing that, we will throw warnings in console saying to change from `size` to `fontSize`;
- In a near future, we will organize a new meet with @ericcleao to discuss breaking changes and refactor this component;

Can you revert the changes you've made on others components, except `packages/yoga/src/Text/native/Text.jsx` and `packages/yoga/src/Text/web/Text.jsx`? You can use this PR or open a new one if you prefer. Please, let us know.

Once again, thank you for the initiative 🫶
When we're done, you'll be added to our [contributors list](https://github.com/gympass/yoga#contributors-).

Cheers,
Caio.

**Comment by oharaju** [comment]
Hi, @caiotracera!

Ok, I got it. I reverted the changes made to other components, and left only the change in the Text component files.

Thanks to @caiotracera , @matheushdsbr and @ericcleao for their attention and review!! 🚀

**Review by matheushdsbr — APPROVED** [review]
Looks great to me o/ 
🚀🚀🚀

**Review by caioalexandrebr — APPROVED** [review]
You noticed something that nobody else noticed. 
Congratulations on the great contribution!! 🚀🚀🚀

**Comment by matheushdsbr** [comment]
@all-contributors please add @oharaju for bug and doc

**Pull request merged**